llama_index/output_parsers/selection.py

```diff
@@ -48,30 +48,24 @@ class Answer(DataClassJsonMixin):
 
 
 class SelectionOutputParser(BaseOutputParser):
-    REQUIRED_KEYS = {"choice", "reason"}
+    REQUIRED_KEYS = frozenset(Answer.__annotations__)
 
     def _filter_dict(self, json_dict: dict) -> dict:
+        """Filter recursively until a dictionary matches all REQUIRED_KEYS."""
         output_dict = json_dict
         for key, val in json_dict.items():
             if key in self.REQUIRED_KEYS:
                 continue
             elif isinstance(val, dict):
-                found = True
-                for key in self.REQUIRED_KEYS:
-                    if key not in val:
-                        found = False
-                        break
-                if found:
-                    output_dict = val
-                    break
+                output_dict = self._filter_dict(val)
             elif isinstance(val, list):
                 for item in val:
                     if isinstance(item, dict):
                         output_dict = self._filter_dict(item)
 
         return output_dict
 
-    def _validate_output(self, output: List[dict]) -> List[dict]:
```

Comment:
Normally, I wouldn't rename an interface, butttt:
- This name was unintuitive, because validations shouldn't be doing conversions/formatting
- Is a private interface, so it's alright